tools/run_parallel.sh

```diff
@@ -39,8 +39,11 @@ echo "  num_iter:        $num_iter"
 outf=./outfile.txt
 
 for i in $(seq $num_iter); do
-  echo "Starting iteration $i" >> $outf
-  echo "Starting iteration $i"
+  echo "==== Starting iteration $i ====" >> $outf
+  echo "==== Starting iteration $i ===="
+
+  echo "Disk usage and free space for the current drive:"
+  df -h .
```

Comment:
I'm thinking if this information would be advantageous? Not many tests are using run_parallel scirpt in our evergreen tests and diagnosing a disk space issue can be quite readily available once we spawn the host. I believe this will be important if we can't gain access to the host. But as of now I believe evergreen testing doesn't run this script on a host that can't be easily accessed (spawnable).

Reply:
As you suggest, spawning a host is a great for diagnosing many bugs, including disk space issues. 

In the case of the recently added stress tests, for which `run_parallel.sh` was originally created, seeing how the disk space evolves through the stress test iterations makes it very easy to see what is happening, both for failing tests and to see what how the disk space evolved compared to on previous runs of that test.

For example, here's part of the log from a patch build looking at the the test that fails in https://jira.mongodb.org/browse/WT-8253 showing how each iteration consumes disk space until it runs out:

`[2021/11/02 21:32:18.044] ==== Starting iteration 1 ====
[2021/11/02 21:32:18.044] Disk usage and free space for the current drive:
[2021/11/02 21:32:18.045] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:32:18.045] /dev/xvdb        50G  2.5G   48G   5% /data
[2021/11/02 21:33:18.642] ==== Starting iteration 2 ====
[2021/11/02 21:33:18.642] Disk usage and free space for the current drive:
[2021/11/02 21:33:18.642] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:33:18.642] /dev/xvdb        50G  4.7G   46G  10% /data
[2021/11/02 21:34:39.819] ==== Starting iteration 3 ====
[2021/11/02 21:34:39.819] Disk usage and free space for the current drive:
[2021/11/02 21:34:39.819] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:34:39.819] /dev/xvdb        50G  6.7G   44G  14% /data
[2021/11/02 21:35:39.121] ==== Starting iteration 4 ====
[2021/11/02 21:35:39.121] Disk usage and free space for the current drive:
[2021/11/02 21:35:39.121] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:35:39.121] /dev/xvdb        50G   11G   40G  22% /data
[2021/11/02 21:36:38.768] ==== Starting iteration 5 ====
[2021/11/02 21:36:38.768] Disk usage and free space for the current drive:
[2021/11/02 21:36:38.768] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:36:38.768] /dev/xvdb        50G   11G   40G  22% /data
[2021/11/02 21:37:37.285] ==== Starting iteration 6 ====
[2021/11/02 21:37:37.285] Disk usage and free space for the current drive:
[2021/11/02 21:37:37.285] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:37:37.285] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:38:39.116] ==== Starting iteration 7 ====
[2021/11/02 21:38:39.116] Disk usage and free space for the current drive:
[2021/11/02 21:38:39.116] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:38:39.116] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:39:37.847] ==== Starting iteration 8 ====
[2021/11/02 21:39:37.847] Disk usage and free space for the current drive:
[2021/11/02 21:39:37.847] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:39:37.847] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:40:57.880] ==== Starting iteration 9 ====
[2021/11/02 21:40:57.880] Disk usage and free space for the current drive:
[2021/11/02 21:40:57.880] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:40:57.880] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:42:01.468] ==== Starting iteration 10 ====
[2021/11/02 21:42:01.468] Disk usage and free space for the current drive:
[2021/11/02 21:42:01.469] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:42:01.469] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:43:16.092] ==== Starting iteration 11 ====
[2021/11/02 21:43:16.092] Disk usage and free space for the current drive:
[2021/11/02 21:43:16.094] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:43:16.094] /dev/xvdb        50G   19G   32G  38% /data
[2021/11/02 21:44:16.631] ==== Starting iteration 12 ====
[2021/11/02 21:44:16.631] Disk usage and free space for the current drive:
[2021/11/02 21:44:16.632] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:44:16.632] /dev/xvdb        50G   23G   28G  46% /data
[2021/11/02 21:45:38.680] ==== Starting iteration 13 ====
[2021/11/02 21:45:38.680] Disk usage and free space for the current drive:
[2021/11/02 21:45:38.681] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:45:38.681] /dev/xvdb        50G   27G   24G  54% /data
[2021/11/02 21:46:37.328] ==== Starting iteration 14 ====
[2021/11/02 21:46:37.328] Disk usage and free space for the current drive:
[2021/11/02 21:46:37.329] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:46:37.329] /dev/xvdb        50G   27G   24G  54% /data
[2021/11/02 21:47:41.895] ==== Starting iteration 15 ====
[2021/11/02 21:47:41.895] Disk usage and free space for the current drive:
[2021/11/02 21:47:41.896] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:47:41.896] /dev/xvdb        50G   27G   24G  54% /data
[2021/11/02 21:48:56.239] ==== Starting iteration 16 ====
[2021/11/02 21:48:56.239] Disk usage and free space for the current drive:
[2021/11/02 21:48:56.240] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:48:56.240] /dev/xvdb        50G   27G   24G  54% /data
[2021/11/02 21:49:56.209] ==== Starting iteration 17 ====
[2021/11/02 21:49:56.209] Disk usage and free space for the current drive:
[2021/11/02 21:49:56.210] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:49:56.210] /dev/xvdb        50G   35G   16G  70% /data
[2021/11/02 21:51:14.187] ==== Starting iteration 18 ====
[2021/11/02 21:51:14.188] Disk usage and free space for the current drive:
[2021/11/02 21:51:14.189] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:51:14.189] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:52:32.252] ==== Starting iteration 19 ====
[2021/11/02 21:52:32.252] Disk usage and free space for the current drive:
[2021/11/02 21:52:32.253] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:52:32.253] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:53:33.866] ==== Starting iteration 20 ====
[2021/11/02 21:53:33.866] Disk usage and free space for the current drive:
[2021/11/02 21:53:33.867] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:53:33.867] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:54:40.915] ==== Starting iteration 21 ====
[2021/11/02 21:54:40.915] Disk usage and free space for the current drive:
[2021/11/02 21:54:40.916] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:54:40.916] /dev/xvdb        50G   43G  7.5G  86% /data
[2021/11/02 21:55:56.711] ==== Starting iteration 22 ====
[2021/11/02 21:55:56.711] Disk usage and free space for the current drive:
[2021/11/02 21:55:56.712] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:55:56.712] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:56:56.176] ==== Starting iteration 23 ====
[2021/11/02 21:56:56.176] Disk usage and free space for the current drive:
[2021/11/02 21:56:56.177] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:56:56.177] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:57:57.389] ==== Starting iteration 24 ====
[2021/11/02 21:57:57.389] Disk usage and free space for the current drive:
[2021/11/02 21:57:57.390] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:57:57.390] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 21:58:59.315] ==== Starting iteration 25 ====
[2021/11/02 21:58:59.315] Disk usage and free space for the current drive:
[2021/11/02 21:58:59.316] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 21:58:59.316] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 22:00:16.096] ==== Starting iteration 26 ====
[2021/11/02 22:00:16.096] Disk usage and free space for the current drive:
[2021/11/02 22:00:16.097] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:00:16.097] /dev/xvdb        50G   43G  7.4G  86% /data
[2021/11/02 22:01:16.187] ==== Starting iteration 27 ====
[2021/11/02 22:01:16.187] Disk usage and free space for the current drive:
[2021/11/02 22:01:16.188] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:01:16.188] /dev/xvdb        50G   44G  6.4G  88% /data
[2021/11/02 22:02:36.295] ==== Starting iteration 28 ====
[2021/11/02 22:02:36.295] Disk usage and free space for the current drive:
[2021/11/02 22:02:36.296] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:02:36.296] /dev/xvdb        50G   47G  3.9G  93% /data
[2021/11/02 22:03:36.898] ==== Starting iteration 29 ====
[2021/11/02 22:03:36.898] Disk usage and free space for the current drive:
[2021/11/02 22:03:36.899] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:03:36.899] /dev/xvdb        50G   49G  1.9G  97% /data
[2021/11/02 22:04:39.050] ==== Starting iteration 30 ====
[2021/11/02 22:04:39.050] Disk usage and free space for the current drive:
[2021/11/02 22:04:39.051] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:04:39.051] /dev/xvdb        50G   49G  1.2G  98% /data
[2021/11/02 22:05:41.074] ==== Starting iteration 31 ====
[2021/11/02 22:05:41.074] Disk usage and free space for the current drive:
[2021/11/02 22:05:41.076] Filesystem      Size  Used Avail Use% Mounted on
[2021/11/02 22:05:41.076] /dev/xvdb        50G   50G  161M 100% /data
[2021/11/02 22:06:28.747] iteration 31 of parallel command 2 failed with 1 error code
[2021/11/02 22:06:28.750] Command failed: command encountered problem: error waiting on process '69d64f19-7600-4d90-99ee-9f1b8a86ab90': exit status 1
[2021/11/02 22:06:28.764] Task completed - FAILURE.`

The fix for https://jira.mongodb.org/browse/WT-8253 will most likely involve reducing the number of iterations (a parameter for each stress test in `evergreen.yml`) for each test to fit within the instance's disk space.

Reply:
I don't have the knowledge of the complete problem, but I got the doubt why the disk usage keeps on reducing? Is the test is not removing the test directories once the test is passed for that iteration?

Reply:
Hi @kommiharibabu , It would appear that the test is not removing the test directories at the end of each test.

Reply:
And that is the same for other stress tests too - it's just that they don't use enough disk space to run out.

Reply:
Thanks, @jeremythorp. The test continues till all the configured iterations or stops whenever it fails. We don't need the test directories once the test has passed. Is the single iteration itself is taking a huge amount of disk space that led to this problem?

Reply:
Good point @kommiharibabu I'm actually curious on this too

Reply:
It turns out that each test (in each iteration) was not deleting the database files at the end, but the subsequent test was replacing them. It was the `nohup.out.*` log files that were being appended to with every test that was consuming all the space.